cloudtrail-check.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+# 
+# This script is here to automate (as much as possible) the tasks in
+# https://cloud.gov/docs/ops/maintenance-list/#review-aws-cloudtrail-events
+# 
+# authorized events will include:
+#   a user name like i-17deadbeef1234567
+#   a source IP address within AWS.
+#   an AWS access key starting with ASIA
+# 
+# XXX The source IP range is kind of janky.  It would be better if this
+#     pulled dynamically from AWS and also if we handled cidr notation
+#     properly.  But these two ranges seem to work for govcloud right now.
+
+if [ "$1" = "-a" ] ; then
+	AUDITMODE=true
+fi
+RETURN=0
+
+START=$(date -v -15d +"%Y-%m-%dT%H:%M:%SZ")
+echo "pulling cloudtrail from a 15 days ago so that 2 week audits will always overlap a bit ($START)"
+
+aws cloudtrail lookup-events --no-paginate --start-time "$START" --lookup-attributes AttributeKey=EventName,AttributeValue=ConsoleLogin > /tmp/consolefoo.$$
+
+echo "============ Console logins per user"
+echo "============ Run 'aws cloudtrail lookup-events --no-paginate --lookup-attributes AttributeKey=EventName,AttributeValue=ConsoleLogin'"
+echo "============ for details on the logins if something seems unusual about the users or the number of times they logged in."
+cat /tmp/consolefoo.$$ | jq -r .Events[].Username | sort | uniq -c
```

Comment:
These couple of lines about `Console logins per user` should be its own script.

Reply:
@LinuxBozo I believe @timothy-spencer original idea with this is taken from this from the docs.

> These EventNames should be attributed to human individuals on the cloud.gov team:
> - `ConsoleLogin`

The idea here is that if we know someone is out of the office, or has had access removed, they should no longer have `ConsoleLogin` appears any more. So it's to be expected that their access should be at `zero` if they're not around. Does that make sense?

I think the value add isn't as high as some of the other metrics we look for while on support, but having it in a pipeline running every ten minutes or every half hour would be more helpful than ignoring the metric completely.

Reply:
I agree.  I think that adding it to a pipeline is not going to be as valuable as people think, because there are subjective parts to the audit, and it's just as easy to run a script as it is to go look at some pipeline output.

That said, it is not a bad thing to do.  Just extra work.  I'm happy to do that.

Reply:
So no one else misses the point going forward: It's not valuable because the limit parameters are not valuable due to not being properly defined. There should be nothing subjective about it. Due to that, the value has absolutely nothing to do with automation or not.

Reply:
Well, the whole point of this audit is that it is something that needs to be done by a human, right?  That's why it's in the tasks we must do when we are on support.  What this script does is automate the gathering of the information that we as operators are required to evaluate, not turn the audit into a "yes/no" script.  Doing that would take FAR more work:  https://xkcd.com/1425/

This is why I think that the utility of creating a pipeline and so on is marginal.  Again, not opposed to doing that.  Just trying to make sure that people were aware that this script may be a solution to a different problem than they are thinking of.  I was trying to solve the problem of "man, paging through all this json takes a lot of time and could be done better by a computer, especially when we have these convenient rules laid out in our documentation".

Reply:
The point of any audit, not just this one, isn't about it must be done by a human. It's that it exists, it has parameters that make sense and that **they can be properly evaluated** in the first place, regardless of whether it's a human or not. This is a blatant hole in the parameters of this particular check. It has nothing to do really with this script, or your work on it, but rather the original context. As such, we shouldn't be replicating something that doesn't make sense, but rather change the parameters as a team, and then re-evaluate scripting it.